zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -202,40 +226,78 @@ impl<F: FieldExt> Transaction<F> {
 
 #[derive(Debug, Default)]
 pub struct Call<F> {
+    /// The unique identifier of call in the whole proof, using the
+    /// `rw_counter` at the call step.
     pub id: usize,
+    /// Indicate if the call is the root call
     pub is_root: bool,
+    /// Indicate if the call is a create call
     pub is_create: bool,
+    /// The identifier of current executed bytecode
     pub opcode_source: F,
+    /// The `rw_counter` at the end of reversion of a call if it has
+    /// `is_persistent == false`
     pub rw_counter_end_of_reversion: usize,
+    /// The call index of caller
     pub caller_call_id: usize,
+    /// The depth in the call stack
     pub depth: usize,
+    /// The caller address
     pub caller_address: Address,
+    /// The callee address
     pub callee_address: Address,
+    /// The call data offset in the memory
     pub call_data_offset: usize,
+    /// The length of call data
     pub call_data_length: usize,
+    /// The return data offset in the memory
     pub return_data_offset: usize,
+    /// The length of return data
     pub return_data_length: usize,
+    /// The ether amount of the transaction
     pub value: Word,
+    /// TBD, Han will update this field
     pub result: Word,
+    /// Indicate if this call and all its caller have `is_success == true`
     pub is_persistent: bool,
```

Comment:
Is `is_persistent` for the caller or for the callee? @han0110

Reply:
It means if state write will be persistent or not, for **each call**. See [this note](https://hackmd.io/OVFxN25cQoKrHCkrHd0Fww) for further explanation about fields `is_success`, `is_persistent` and `rw_counter_end_of_reversion`.

Reply:
Thanks for the explanation, the document makes sense to me now.